zulipterminal/ui_tools/messages.py

```diff
@@ -715,7 +721,11 @@ def main_view(self) -> List[Any]:
 
         # Transform raw message content into markup (As needed by urwid.Text)
         content, self.message_links, self.time_mentions = self.transform_content(
-            self.message["content"], self.model.server_url
+            self.message["content"],
+            self.model.server_url,
+            self.alerted_words,
+            "has_alert_word" in self.message["flags"]
```

Comment:
These new parameters could be combined into one. If the message doesn't have alert words flagged, then the function doesn't need to know/process them.